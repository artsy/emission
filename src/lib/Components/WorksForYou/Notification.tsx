import * as React from "react"
import { Image, StyleSheet, TextStyle, TouchableWithoutFeedback, View, ViewStyle } from "react-native"
import * as Relay from "react-relay"

import SwitchBoard from "../../NativeModules/SwitchBoard"
import ArtworksGrid from "../ArtworkGrids/GenericGrid"
import Headline from "../Text/Headline"
import SerifText from "../Text/Serif"

import colors from "../../../data/colors"

export class Notification extends React.Component<RelayProps, any> {
  handleArtistTap() {
    // Special notifications will pass down an artistHref. Otherwise, grab it from the artworks
    const artistHref = this.props.notification.artistHref || this.props.notification.artworks[0].artists[0].href
    if (!artistHref) {
      return
    }
    SwitchBoard.presentNavigationViewController(this, artistHref)
  }

  render() {
    const notification = this.props.notification

    // artwork-less notifications are rare but possible and very unsightly
    if (!notification.artworks.length) {
      return null
    }

    return (
      <View style={styles.container}>
        <TouchableWithoutFeedback onPress={this.handleArtistTap.bind(this)}>
          <View style={styles.header}>
            {notification.image &&
              <Image source={{ uri: notification.image.resized.url }} style={styles.artistAvatar} />}
            <View style={styles.metadataContainer}>
              <View style={styles.nameAndStatusContainer}>
                <Headline style={styles.artistName}>{notification.artists}</Headline>
                {notification.status === "UNREAD" && <View style={styles.readStatus} />}
              </View>
              <SerifText style={styles.metadata}>{notification.message + " Â· " + notification.date}</SerifText>
            </View>
          </View>
        </TouchableWithoutFeedback>
        <View style={styles.gridContainer}>
          <ArtworksGrid artworks={notification.artworks} />
        </View>
      </View>
    )
  }
}

interface Styles {
  container: ViewStyle
  header: ViewStyle
  artistAvatar: ViewStyle
  metadataContainer: ViewStyle
  nameAndStatusContainer: ViewStyle
  readStatus: ViewStyle
  artistName: TextStyle
  metadata: TextStyle
  gridContainer: ViewStyle
}

const styles = StyleSheet.create<Styles>({
  container: {
    marginTop: 20,
  },
  header: {
    flexDirection: "row",
  },
  artistAvatar: {
    height: 40,
    width: 40,
    backgroundColor: colors["gray-light"],
    alignSelf: "center",
    borderRadius: 20,
    marginRight: 10,
  },
  metadataContainer: {
    alignSelf: "center",
    flex: 1,
  },
  nameAndStatusContainer: {
    flexDirection: "row",
    justifyContent: "space-between",
    flexGrow: 1,
  },
  readStatus: {
    backgroundColor: colors["purple-regular"],
    width: 10,
    height: 10,
    borderRadius: 5,
    alignSelf: "center",
  },
  artistName: {
    fontSize: 14,
  },
  metadata: {
    marginTop: 2,
    fontSize: 16,
    color: colors["gray-semibold"],
  },
  gridContainer: {
    marginTop: 20,
    marginBottom: 20,
  },
})

export default Relay.createContainer(Notification, {
  fragments: {
    notification: () => Relay.QL`
      fragment on NotificationsFeedItem {
        date(format: "MMM D")
        message
        artists
        artworks {
          artists(shallow: true) {
            href
          }
          ${ArtworksGrid.getFragment("artworks")}
        }
        status
        image {
          resized(height: 80, width: 80) {
            url
          }
        }
      }
    `,
  },
})

interface RelayProps {
  notification: {
    date: string
    message: string
    artists: string
    artworks: any[]
    status: string
    image: {
      resized: {
        url: string
      }
    } | null
    artistHref?: string | null
  }
}
