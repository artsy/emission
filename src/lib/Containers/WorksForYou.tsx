import React from "react"
import { ConnectionData, createPaginationContainer, graphql, RelayPaginationProp } from "react-relay"

import {
  ListView,
  ListViewDataSource,
  NativeModules,
  RefreshControl,
  ScrollView,
  StyleSheet,
  TextStyle,
  View,
  ViewStyle,
} from "react-native"

import Events from "../NativeModules/Events"

import { PAGE_SIZE } from "lib/data/constants"

import ZeroState from "lib/Components/States/ZeroState"
import Notification from "lib/Components/WorksForYou/Notification"
import colors from "lib/data/colors"
import { isCloseToBottom } from "lib/utils/isCloseToBottom"

import { WorksForYou_viewer } from "__generated__/WorksForYou_viewer.graphql"

interface Props {
  relay: RelayPaginationProp
  viewer: WorksForYou_viewer
}

interface State {
  dataSource: ListViewDataSource | null
  isRefreshing: boolean
}

export class WorksForYou extends React.Component<Props, State> {
  // TODO: This `| any` is a hack workaround to a typing bug in https://github.com/artsy/emission/pull/504/
  scrollView?: ScrollView | any
  currentScrollOffset?: number = 0

  constructor(props: Props) {
    super(props)

    const notifications: object[] = this.props.viewer.me.followsAndSaves.notifications.edges.map(edge => edge.node)
    if (this.props.viewer.selectedArtist) {
      notifications.unshift(this.formattedSpecialNotification())
    }

    const dataSource =
      notifications.length &&
      new ListView.DataSource({ rowHasChanged: (row1, row2) => row1 !== row2 }).cloneWithRows(notifications)

    this.state = {
      dataSource,
      isRefreshing: false,
    }
  }

  componentDidMount() {
    // Update read status in gravity
    NativeModules.ARTemporaryAPIModule.markNotificationsRead(error => {
      if (error) {
        console.warn(error)
      } else {
        Events.postEvent({
          name: "Notifications read",
          source_screen: "works for you page",
        })
      }
    })
  }

  formattedSpecialNotification() {
    const artist = this.props.viewer.selectedArtist

    return {
      // This is just some unique ID, don’t rely on MP being able to retrieve a notification by this ID.
      __id: `notification-${artist.id}`,
      message: artist.artworks.length + (artist.artworks.length > 1 ? " Works Added" : " Work Added"),
      artists: artist.name,
      artworks: artist.artworks,
      image: {
        resized: {
          url: artist.image.resized.url,
        },
      },
      artistHref: artist.href,
    }
  }

  fetchNextPage = () => {
    if (!this.props.relay.hasMore() || this.props.relay.isLoading()) {
      return
    }
    this.props.relay.loadMore(PAGE_SIZE, error => {
      if (error) {
        // FIXME: Handle error
        console.error("WorksForYou.tsx", error.message)
      }

      const notifications: object[] = this.props.viewer.me.followsAndSaves.notifications.edges.map(edge => edge.node)

      // Make sure we maintain the special notification if it exists
      if (this.props.viewer.selectedArtist) {
        notifications.unshift(this.formattedSpecialNotification())
      }

      this.setState({
        dataSource: this.state.dataSource.cloneWithRows(notifications),
      })
    })
  }

  handleRefresh = () => {
    this.setState({ isRefreshing: true })
    this.props.relay.refetchConnection(PAGE_SIZE, error => {
      if (error) {
        // FIXME: Handle error
        console.error("WorksForYou.tsx #handleRefresh", error.message)
      }
      this.setState({ isRefreshing: false })
    })
  }

  render() {
    const hasNotifications = this.state.dataSource

    /* If showing the empty state, the ScrollView should have a {flex: 1} style so it can expand to fit the screen.
       otherwise, it should not use any flex growth.
    */
    return (
      <ScrollView
        contentContainerStyle={hasNotifications ? {} : styles.container}
        onScroll={isCloseToBottom(this.fetchNextPage)}
        scrollEventThrottle={100}
        ref={scrollView => (this.scrollView = scrollView)}
        refreshControl={<RefreshControl refreshing={this.state.isRefreshing} onRefresh={this.handleRefresh} />}
      >
        <View style={{ flex: 1 }}>{hasNotifications ? this.renderNotifications() : this.renderEmptyState()}</View>
      </ScrollView>
    )
  }

  renderNotifications() {
    return (
      <ListView
        dataSource={this.state.dataSource}
        renderRow={data => <Notification notification={data} />}
        renderSeparator={(sectionID, rowID) =>
          <View key={`${sectionID}-${rowID}`} style={styles.separator} /> as React.ReactElement<{}>
        }
        scrollEnabled={false}
      />
    )
  }

  renderEmptyState() {
    return (
      <ZeroState
        title="You haven’t followed any artists yet."
        subtitle="Follow artists to see new works that have been added to Artsy"
      />
    )
  }
}

interface Styles {
  container: ViewStyle
  title: TextStyle
  separator: ViewStyle
}

const styles = StyleSheet.create<Styles>({
  container: {
    flex: 1,
  },
  title: {
    marginTop: 20,
    fontSize: 20,
  },
  separator: {
    height: 1,
    backgroundColor: colors["gray-regular"],
  },
})

const WorksForYouContainer = createPaginationContainer(
  WorksForYou,
  {
    viewer: graphql`
      fragment WorksForYou_viewer on Viewer
        @argumentDefinitions(
          count: { type: "Int", defaultValue: 10 }
          cursor: { type: "String" }
          selectedArtist: { type: "String!", defaultValue: "" }
          sort: { type: "ArtworkSorts" }
        ) {
        me {
          followsAndSaves {
            notifications: bundledArtworksByArtist(sort: PUBLISHED_AT_DESC, first: $count, after: $cursor)
              @connection(key: "WorksForYou_notifications") {
              pageInfo {
                hasNextPage
                endCursor
              }
              edges {
                node {
                  __id
                  ...Notification_notification
                }
              }
            }
          }
        }
        selectedArtist: artist(id: $selectedArtist) {
          id
          href
          name
          image {
            resized(height: 80, width: 80) {
              url
            }
          }
          artworks(sort: published_at_desc, size: 6) {
            ...GenericGrid_artworks
          }
        }
      }
    `,
  },
  {
    direction: "forward",
    getConnectionFromProps(props) {
      return props.viewer.me.followsAndSaves.notifications as ConnectionData
    },
    getFragmentVariables(prevVars, totalCount) {
      return {
        ...prevVars,
        count: totalCount,
      }
    },
    getVariables(_props, { count, cursor }, fragmentVariables) {
      return {
        // in most cases, for variables other than connection filters like
        // `first`, `after`, etc. you may want to use the previous values.
        ...fragmentVariables,
        count,
        cursor,
      }
    },
    query: graphql`
      query WorksForYouQuery($count: Int!, $cursor: String) {
        viewer {
          ...WorksForYou_viewer @arguments(count: $count, cursor: $cursor)
        }
      }
    `,
  }
)

export default WorksForYouContainer
