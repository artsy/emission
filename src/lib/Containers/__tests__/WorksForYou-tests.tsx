import * as moment from "moment"
import * as React from "react"
import { NativeModules } from "react-native"
import * as renderer from "react-test-renderer"

import { renderWithLayout } from "../../tests/renderWithLayout"
import { WorksForYou } from "../WorksForYou"

beforeAll(() => {
  NativeModules.ARTemporaryAPIModule = { markNotificationsRead: jest.fn() }
  NativeModules.ARWorksForYouModule = { updateNotificationsCount: jest.fn() }
  WorksForYou.prototype.componentDidUpdate = () => {
    return null
  }
})

describe("with notifications", () => {
  it("creates a ListViewDataSource upon instantiation", () => {
    const worksForYou = new WorksForYou(notificationsResponse())
    expect(worksForYou.state.dataSource).toBeTruthy()
  })

  it("updates the notification count", () => {
    const viewer = notificationsResponse().viewer
    renderer.create(<WorksForYou viewer={viewer} relay={null} />).toJSON()
    expect(NativeModules.ARTemporaryAPIModule.markNotificationsRead).toBeCalled()
  })

  it("lays out correctly on small screens", () => {
    const viewer = notificationsResponse().viewer
    const component = renderWithLayout(<WorksForYou viewer={viewer} relay={null} />, { width: 100 })
    expect(component).toMatchSnapshot()
  })

  it("lays out correctly on larger screens", () => {
    const viewer = notificationsResponse().viewer
    const component = renderWithLayout(<WorksForYou viewer={viewer} relay={null} />, { width: 700 })
    expect(component).toMatchSnapshot()
  })
})

describe("when it has a special notification", () => {
  it("properly formats it and adds it to the top of the dataSource blob", () => {
    const response = selectedArtistResponse()
    const worksForYou = new WorksForYou(response)
    const expectedFormattedNotification = {
      date: moment().format("MMM DD"),
      message: "1 Work Added",
      artists: "Juliana Huxtable",
      artworks: selectedArtistResponse().viewer.selectedArtist.artworks,
      status: "UNREAD",
      image: {
        resized: {
          url: "cloudfront.url",
        },
      },
      artistHref: "artist/juliana-huxtable",
    }
    expect(worksForYou.state.dataSource.getRowData(0, 0)).toEqual(expectedFormattedNotification)
  })
})

describe("without notifications", () => {
  it("does not create a ListViewDataSource", () => {
    const worksForYou = new WorksForYou(emptyStateResponse())
    expect(worksForYou.state.dataSource).toBeFalsy()
  })

  it("lays out correctly on small screens", () => {
    const viewer = emptyStateResponse().viewer
    const component = renderWithLayout(<WorksForYou viewer={viewer} relay={null} />, { width: 100 })
    expect(component).toMatchSnapshot()
  })

  it("lays out correctly on larger screens", () => {
    const viewer = emptyStateResponse().viewer
    const component = renderWithLayout(<WorksForYou viewer={viewer} relay={null} />, { width: 700 })
    expect(component).toMatchSnapshot()
  })
})

interface NotificationsResponse {
  viewer: {
    me: {
      notifications: {
        pageInfo: {
          hasNextPage: boolean
        }
        edges: Array<
          {
            node: {
              artists: string
              date: string
              message: string
              artworks: [{ title: string }]
              image: {
                resized: {
                  url: string
                }
              }
            }
          }
        >
      }
    }

    selectedArtist?: any
  }
}

const notificationsResponse = () => {
  return {
    viewer: {
      me: {
        notifications: {
          pageInfo: {
            hasNextPage: true,
          },
          edges: [
            {
              node: {
                artists: "Jean-Michel Basquiat",
                date: "Mar 16",
                message: "1 Work Added",
                artworks: [{ title: "Anti-Product Postcard" }],
                image: {
                  resized: {
                    url: "cloudfront.url",
                  },
                },
              },
            },
            {
              node: {
                artists: "Ana Mendieta",
                date: "Mar 16",
                message: "2 Works Added",
                artworks: [{ title: "CorazÃ³n de Roca con Sangre" }, { title: "Butterfly" }],
                image: {
                  resized: {
                    url: "cloudfront.url",
                  },
                },
              },
            },
          ],
        },
      },
    },
  } as NotificationsResponse
}

const emptyStateResponse = () => {
  return {
    viewer: {
      me: {
        notifications: {
          pageInfo: {
            hasNextPage: true,
          },
          edges: [],
        },
      },
    },
  }
}

const selectedArtistResponse = () => {
  {
    const response = notificationsResponse()
    response.viewer.selectedArtist = {
      name: "Juliana Huxtable",
      href: "artist/juliana-huxtable",
      image: {
        resized: {
          url: "cloudfront.url",
        },
      },
      artworks: [
        {
          __id: "4594385943",
          title: "Untitled (Casual Power)",
        },
      ],
    }
    return response
  }
}
