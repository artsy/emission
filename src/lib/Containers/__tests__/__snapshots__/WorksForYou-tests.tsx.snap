// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with notifications lays out correctly on larger screens 1`] = `
<RCTScrollView
  contentContainerStyle={Object {}}
  onScroll={[Function]}
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={false}
    />
  }
  scrollEventThrottle={100}
>
  <RCTRefreshControl />
  <View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RCTScrollView
        dataSource={
          ListViewDataSource {
            "items": 2,
          }
        }
        renderRow={[Function]}
        renderScrollComponent={[Function]}
        renderSeparator={[Function]}
        scrollEnabled={false}
      >
        <View>
          <View
            style={
              Object {
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 20,
              }
            }
          >
            <View
              accessible={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "cloudfront.url",
                  }
                }
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "#f8f8f8",
                    "borderRadius": 20,
                    "height": 40,
                    "marginRight": 10,
                    "width": 40,
                  }
                }
              />
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "flex": 1,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontSize": 12,
                      },
                      Object {
                        "fontFamily": "AvantGardeGothicITC",
                      },
                      Object {
                        "fontSize": 12,
                      },
                    ]
                  }
                >
                  JEAN-MICHEL BASQUIAT
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "fontSize": 17,
                      },
                      Object {
                        "color": "#666666",
                        "fontSize": 14,
                        "marginTop": 2,
                      },
                      Object {
                        "fontFamily": "AGaramondPro-Regular",
                      },
                    ]
                  }
                >
                  1 Work Added
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "marginBottom": 20,
                  "marginTop": 20,
                }
              }
            >
              <View
                onLayout={[Function]}
              >
                <View
                  accessibilityLabel="Artworks Content View"
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 20,
              }
            }
          >
            <View
              accessible={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "cloudfront.url",
                  }
                }
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "#f8f8f8",
                    "borderRadius": 20,
                    "height": 40,
                    "marginRight": 10,
                    "width": 40,
                  }
                }
              />
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "flex": 1,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontSize": 12,
                      },
                      Object {
                        "fontFamily": "AvantGardeGothicITC",
                      },
                      Object {
                        "fontSize": 12,
                      },
                    ]
                  }
                >
                  ANA MENDIETA
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "fontSize": 17,
                      },
                      Object {
                        "color": "#666666",
                        "fontSize": 14,
                        "marginTop": 2,
                      },
                      Object {
                        "fontFamily": "AGaramondPro-Regular",
                      },
                    ]
                  }
                >
                  2 Works Added
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "marginBottom": 20,
                  "marginTop": 20,
                }
              }
            >
              <View
                onLayout={[Function]}
              >
                <View
                  accessibilityLabel="Artworks Content View"
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</RCTScrollView>
`;

exports[`with notifications lays out correctly on small screens 1`] = `
<RCTScrollView
  contentContainerStyle={Object {}}
  onScroll={[Function]}
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={false}
    />
  }
  scrollEventThrottle={100}
>
  <RCTRefreshControl />
  <View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RCTScrollView
        dataSource={
          ListViewDataSource {
            "items": 2,
          }
        }
        renderRow={[Function]}
        renderScrollComponent={[Function]}
        renderSeparator={[Function]}
        scrollEnabled={false}
      >
        <View>
          <View
            style={
              Object {
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 20,
              }
            }
          >
            <View
              accessible={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "cloudfront.url",
                  }
                }
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "#f8f8f8",
                    "borderRadius": 20,
                    "height": 40,
                    "marginRight": 10,
                    "width": 40,
                  }
                }
              />
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "flex": 1,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontSize": 12,
                      },
                      Object {
                        "fontFamily": "AvantGardeGothicITC",
                      },
                      Object {
                        "fontSize": 12,
                      },
                    ]
                  }
                >
                  JEAN-MICHEL BASQUIAT
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "fontSize": 17,
                      },
                      Object {
                        "color": "#666666",
                        "fontSize": 14,
                        "marginTop": 2,
                      },
                      Object {
                        "fontFamily": "AGaramondPro-Regular",
                      },
                    ]
                  }
                >
                  1 Work Added
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "marginBottom": 20,
                  "marginTop": 20,
                }
              }
            >
              <View
                onLayout={[Function]}
              >
                <View
                  accessibilityLabel="Artworks Content View"
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 20,
              }
            }
          >
            <View
              accessible={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "cloudfront.url",
                  }
                }
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "#f8f8f8",
                    "borderRadius": 20,
                    "height": 40,
                    "marginRight": 10,
                    "width": 40,
                  }
                }
              />
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "flex": 1,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontSize": 12,
                      },
                      Object {
                        "fontFamily": "AvantGardeGothicITC",
                      },
                      Object {
                        "fontSize": 12,
                      },
                    ]
                  }
                >
                  ANA MENDIETA
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "fontSize": 17,
                      },
                      Object {
                        "color": "#666666",
                        "fontSize": 14,
                        "marginTop": 2,
                      },
                      Object {
                        "fontFamily": "AGaramondPro-Regular",
                      },
                    ]
                  }
                >
                  2 Works Added
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "marginBottom": 20,
                  "marginTop": 20,
                }
              }
            >
              <View
                onLayout={[Function]}
              >
                <View
                  accessibilityLabel="Artworks Content View"
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</RCTScrollView>
`;

exports[`without notifications lays out correctly on larger screens 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flex": 1,
    }
  }
  onScroll={[Function]}
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={false}
    />
  }
  scrollEventThrottle={100}
>
  <RCTRefreshControl />
  <View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "height": "100%",
              "justifyContent": "center",
              "marginLeft": 20,
              "marginRight": 20,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "maxWidth": 320,
                "paddingBottom": 60,
              },
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 30,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            You haven’t followed any artists yet.
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={0}
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 18,
                  "marginTop": 10,
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Follow artists to see new works that have been added to Artsy
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`without notifications lays out correctly on small screens 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flex": 1,
    }
  }
  onScroll={[Function]}
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={false}
    />
  }
  scrollEventThrottle={100}
>
  <RCTRefreshControl />
  <View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "height": "100%",
              "justifyContent": "center",
              "marginLeft": 20,
              "marginRight": 20,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "maxWidth": 320,
                "paddingBottom": 60,
              },
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 30,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            You haven’t followed any artists yet.
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={0}
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 18,
                  "marginTop": 10,
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Follow artists to see new works that have been added to Artsy
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
